Sources/Hyperspace/HTTP/HTTP.swift

```diff
@@ -127,53 +127,93 @@ public struct HTTP {
             self.init(data)
         }
     }
+
+    /// Represents an HTTP request
+    public struct Request: Equatable {
+
+        /// The `URL` to which the `Request` originated.
+        public let url: URL?
+
+        /// The HTTP header fields for this request.
+        public let headers: [String: String]?
+
+        /// The raw `Data` associated with the HTTP request, if any was provided.
+        public let body: Data?
+
+        /// Initialize a new `Request` with any given `URL`, HTTP headers and body.
+        /// - Parameters:
+        ///   - url: The `URL` for which this request was created.
+        ///   - data: The raw `Data` associated with the HTTP request, if any was provided.
+        ///   - headers: The HTTP header fields for this request.
+        public init(url: URL? = nil, headers: [String: String]? = nil, body: Data? = nil) {
```

Comment:
Should we use `[HTTP.HeaderKey: HTTP.HeaderValue]` for the type of the `headers` parameter?

Reply:
We can, we'd have to `map` from the `URLRequest.allHTTPHeaderFields`, but we can. We'd probably want to do the same from the `HTTPURLResponse` on the other side as well. Thoughts?

Reply:
It might be okay to leave them as `[String: String]` for now (whatever most closely maps to the `URLRequest`). We can revisit if we feel we need the abstraction in practice.